### DISCUSSION TITLE
hyper why not support get request when send request failed becase of connection error?

### DISCUSSION THREAD

**Discussion by xj524598** [hyperium/hyper#3671] [Q&A]
in some times  wo need retry becase of connection error ,so why hyper not support get request when send request failed becase of connection error?

i know hyperclient has retry_canceled_requests, seems it can not proccess all the situtation

**Comment by seanmonstar** [comment]
hyper doesn't _automatically_ retry almost any error, since it's not obvious whether doing so is "safe" for the server. If hyper wrote a request to the connection, and then it reports an error, we can't know generally if the server saw any of the request, if it started to act on it, and if the request had side effects.

However, the user of hyper may have better knowledge of that. If you know that the request is safe to retry, you can send it again. You could look at `tower::retry` for middleware to help do that.
